includes/embeds/class-amp-core-block-handler.php

```diff
@@ -45,25 +44,35 @@ public function register_embed() {
 	 * Unregister embed.
 	 */
 	public function unregister_embed() {
-		if ( class_exists( 'WP_Block_Type_Registry' ) ) {
-			$registry = WP_Block_Type_Registry::get_instance();
-			$block    = $registry->get_registered( $this->block_name );
+		if ( ! class_exists( 'WP_Block_Type_Registry' ) ) {
+			return;
+		}
 
-			if ( $block && ! empty( $this->original_categories_callback ) ) {
-				$block->render_callback             = $this->original_categories_callback;
-				$this->original_categories_callback = null;
+		$registry = WP_Block_Type_Registry::get_instance();
+		foreach ( array_keys( $this->block_render_methods ) as $block_name ) {
+			$block = $registry->get_registered( $block_name );
+			if ( $block && isset( $block->original_render_callback ) ) {
+				$block->render_callback = $block->original_render_callback;
 			}
 		}
 	}
 
 	/**
-	 * Render Gutenberg block. This is essentially the same method as the original.
-	 * Difference is excluding the disallowed JS script, adding <form> tags, and using on:change for <select>.
+	 * Render categories block when displayAsDropdown.
+	 *
+	 * This excludes the disallowed JS scrips, adds <form> tags, and uses on:change for <select>.
+	 *
+	 * @see render_block_core_categories()
 	 *
 	 * @param array $attributes Attributes.
 	 * @return string Rendered.
 	 */
-	public function render( $attributes ) {
+	public function render_categories_block( $attributes ) {
+		$archives_block = WP_Block_Type_Registry::get_instance()->get_registered( 'core/categories' );
```

Comment:
Should it be `core/archives`?

Reply:
Yes, but I've totally refactored it now so this is fixed.